osm2gtfs/tests/core/tests_creator_factory.py

```diff
@@ -0,0 +1,92 @@
+
+            # Loop through available directories of creators
+            for selector in dirs:
+
+                # Check if creator directory fits naming convention
+                self.assertTrue(
+                    self._check_naming_convention(selector),
+                    'The path "creators/' + selector +
+                    ' " doesn\'t fit the naming convention.')
+
+                # Check for existing configuration file
+                config_file_path = path + selector + "/" + selector + ".json"
+                self.assertTrue(
+                    os.path.isfile(config_file_path),
+                    'No correctly named configuration file found for: ' +
+                    selector)
+
+                # Check whether selector in config file is correctly
+                config = Configuration.load_config_file(open(config_file_path))
+                self.assertEqual(
+                    selector, config['selector'],
+                    "The selector (" + config['selector'] +
+                    ") in config file is not equal to the directory name: " +
+                    selector)
+
+                # Loop through available files in creator directories and check
+                # whether they are correctly named
+                for inner_subdir, inner_dirs, inner_files in os.walk(
+                        path + selector):
+
+                    for filename in inner_files:
+                        # Check the pyhton files (and ignore the others)
+                        if filename.endswith(
+                                '.py') and not filename == "__init__.py":
+
+                            # Check if selector has been properly applied to
+                            # file name
+                            self.assertTrue(
+                                filename.endswith(selector + '.py'),
+                                "This file is not well named: " + filename)
+
+                            # Check for valid creators
+                            split_filename = filename.split("_")
+                            self.assertTrue(
+                                split_filename[0] in creators,
+                                "This file is not well named: " + filename)
+                            self.assertEqual(
+                                split_filename[1], "creator",
+                                "This file is not well named: " + filename)
+
+    def _check_naming_convention(self, selector):
+        check = True
+        if "_" in selector:
+            split_selector = selector.split("_")
+
+            # Check for ISO 3166-2 country code
+            if not len(split_selector[0]) == 2:
+                check = False
+
+            # Check for amount of split elements; can only be 2 or three
+            if len(split_selector) < 2 or len(split_selector) > 3:
+                check = False
+        else:
+            return False
+        return check
```

Comment:
Generally, +1 for adding this test!

Correct me if I'm wrong but this only tests the filenames and not the class names as required by the `CreatorsFactory`. Not sure how difficult it would be to implement this tests too.

Reply:
Added the desired test.